executor/src/witgen/machines/machine_extractor.rs

```diff
@@ -118,12 +118,32 @@ pub fn split_out_machines<'a, T: FieldElement>(
             log::info!("Detected machine: block");
             machines.push(KnownMachine::BlockMachine(machine));
         } else {
-            log::info!("Could not detect a specific machine. Will use the generic VM machine.");
+            log::info!("Detected machine: VM.");
+            let latch = connecting_identities
+                .iter()
+                .fold(None, |existing_latch, identity| {
+                    let current_latch = identity
+                        .right
+                        .selector
+                        .as_ref()
+                        .expect("Expected secondary VMs to have a latch");
```

Comment:
Is that a fair requirement? What about a dynamic VM?

Reply:
It can be dynamic, if the VM is compiled from Powdr ASM, it will always be the `instr_return` register!

Here, it would only panic of you did something like `selector {a, b, c} in {d, e, f}` (with no selector on the RHS) where the RHS is a VM. That wouldn't be a useful VM though.

Reply:
ah ok

Reply:
But should we really panic in that case? Isn't it more like "witgen failed"? We could also just still try, maybe the LHS selector is always zero.

Reply:
Hmm, IDK, I'd have to see an actual use case to see what to do in that situation. A VM that can be queried in every row does not make sense IMO, so I'd rather fail early here. If this gets detected as a VM and doesn't have a latch, it probably should not have been detected as a VM, so it makes sense to me that it panics in the machine extractor.

Reply:
But this is the "else" case of machine detection, so it's not really detected as a VM, it's just not detected as anything else.

Reply:
I changed the error message.